src/util/shmem/fd_shmem.h

```diff
@@ -269,6 +269,19 @@ int
 fd_shmem_leave_anonymous( void *                 join,
                           fd_shmem_join_info_t * opt_info );
 
+/* fd_shmem_grab_region will attempt to map a region at the passed
+   address with the passed size. If the return value of `mmap` equals
+   the passed address this means the area of memory was unmapped
+   previously and we havesuccesfully "grabbed" the region. We can then
+   call `mmap` with MAP_FIXED over the region and be certain no
+   corruption occurs. If the return value of `mmap` does not return the
+   passed address this means that the passed region is already at least
+   partially mapped and we cannot grab it. */
+
+int
+fd_shmem_grab_region( ulong addr,
```

Comment:
This isn't really an interface on fd_shmem, I would not expose this, and leave the code in util unchanged.  You can just copy paste the grabber code into fd_topo it's a bit cleaner for now.